drivers/fuel_gauge/sy24561/sy24561.c

```diff
@@ -0,0 +1,346 @@
+static int sy24561_set_status(const struct device *dev, uint16_t const status)
+{
+	LOG_DBG("Setting status to %" PRIu16, status);
+
+	if (status != 0) {
+		LOG_ERR("Invalid status %" PRIu16 ", it should be 0", status);
+		return -EINVAL;
+	}
+
+	uint16_t config;
+	int const ret = sy24561_get_config(dev, &config);
+
+	if (ret != 0) {
+		LOG_ERR("Failed to read config: %d", ret);
+		return ret;
+	}
+
+	LOG_DBG("config register: 0x%x", config);
+
+	/* This comes from datasheet at page 6 */
+	uint16_t const status_mask = 1 << 5;
+
+	config &= ~status_mask;
+	LOG_DBG("new config register: 0x%x", config);
+
+	if (sy24561_write_reg(dev, SY24561_REG_CONFIG, config) != 0) {
+		LOG_ERR("Impossible to write config register");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int sy24561_set_alarm_threshold(const struct device *dev, uint16_t const percent_threshold)
+{
+	LOG_DBG("Setting SOC alarm threshold to %" PRIu16, percent_threshold);
+
+	uint16_t config;
+	int const ret = sy24561_get_config(dev, &config);
+
+	if (ret != 0) {
+		LOG_ERR("Failed to read config: %d", ret);
+		return ret;
+	}
+
+	LOG_DBG("config register: 0x%x", config);
+
+	/* This comes from datasheet at page 6 */
+	uint16_t const percent_threshold_max = 32;
+	uint16_t const percent_threshold_min = 1;
+
+	if (percent_threshold > percent_threshold_max) {
+		LOG_ERR("SOC alarm threshold %" PRIu16 " should be <= %" PRIu16, percent_threshold,
+			percent_threshold_max);
+		return -EINVAL;
+	}
+
+	if (percent_threshold < percent_threshold_min) {
+		LOG_ERR("SOC alarm threshold %" PRIu16 " should be >= %" PRIu16, percent_threshold,
+			percent_threshold_min);
+		return -EINVAL;
+	}
+
+	/* This comes from datasheet at page 6 */
+	uint16_t const alarm_threshold_mask = 0b11111;
+	uint16_t const alarm_threshold = 32 - percent_threshold;
+
+	config &= ~alarm_threshold_mask;
+	config |= alarm_threshold;
+	LOG_DBG("new config register: 0x%x", config);
+
+	if (sy24561_write_reg(dev, SY24561_REG_CONFIG, config) != 0) {
+		LOG_ERR("Impossible to write config register");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int sy24561_set_temperature(const struct device *dev, uint16_t temperature_dK /* 0.1K */)
+{
+	LOG_DBG("Setting temperature to %" PRIdK(u16), dKARGS(temperature_dK));
+
+	uint16_t config;
+	int const ret = sy24561_get_config(dev, &config);
+
+	if (ret != 0) {
+		LOG_ERR("Failed to read config: %d", ret);
+		return ret;
+	}
+
+	/* This comes from datasheet at page 5 */
+	uint16_t const temperature_dK_max = CELSIUS_TO_DECI_KELVIN(60);
+	uint16_t const temperature_dK_min = CELSIUS_TO_DECI_KELVIN(-20);
+
+	if (temperature_dK > temperature_dK_max) {
+		LOG_ERR("Temperature %" PRIdK(u16) " should be <= %" PRIdK(u16),
+			dKARGS(temperature_dK), dKARGS(temperature_dK_max));
+		return -EINVAL;
+	}
+
+	if (temperature_dK < temperature_dK_min) {
+		LOG_ERR("Temperature %" PRIdK(u16) " should be >= %" PRIdK(u16),
+			dKARGS(temperature_dK), dKARGS(temperature_dK_min));
+		return -EINVAL;
+	}
```

Comment:
Same as in `sy24561_set_alarm_threshold()`
As mentioned in the last review:  
> In this case it's better to throw a warning and compensate, instead of throwing an error


```suggestion
	if (temperature_dK > temperature_dK_max) {
		LOG_WRN("Temperature %" PRIdK(u16) " should be <= %" PRIdK(u16),
			dKARGS(temperature_dK), dKARGS(temperature_dK_max));
		temperature_dK = temperature_dK_max;
	}

	if (temperature_dK < temperature_dK_min) {
		LOG_WRN("Temperature %" PRIdK(u16) " should be >= %" PRIdK(u16),
			dKARGS(temperature_dK), dKARGS(temperature_dK_min));
		temperature_dK = temperature_dK_min;
	}
```